clients/python/tests/test_api.py

```diff
@@ -56,19 +59,146 @@ def test_post_model_version_artifacts(auth_headers: dict, artifact_type: str, ur
+    """
+    # First, create an experiment
+    experiment_payload = {
+        "name": f"test-experiment-{secrets.randbelow(1000000)}",
+        "externalId": str(secrets.randbelow(2000000000 - 100000 + 1) + 100000),
+        "description": "Test experiment for artifact testing"
+    }
+    exp_response = requests.post(
+        f"{REGISTRY_URL}/api/model_registry/v1alpha3/experiments",
+        headers=auth_headers,
+        json=experiment_payload,
+        timeout=DEFAULT_API_TIMEOUT
+    )
+    assert exp_response.status_code in {200, 201}, f"Failed to create experiment: {exp_response.text}"
+    experiment_id = exp_response.json()["id"]
+    
+    # Then, create an experiment run for that experiment
+    experiment_run_payload = {
+        "experimentId": experiment_id,
+        "name": f"test-run-{secrets.randbelow(1000000)}",
+        "externalId": str(secrets.randbelow(2000000000 - 100000 + 1) + 100000),
+        "description": "Test experiment run for artifact testing"
+    }
+    run_response = requests.post(
+        f"{REGISTRY_URL}/api/model_registry/v1alpha3/experiment_runs",
+        headers=auth_headers,
+        json=experiment_run_payload,
+        timeout=DEFAULT_API_TIMEOUT
+    )
+    assert run_response.status_code in {200, 201}, f"Failed to create experiment run: {run_response.text}"
+    experiment_run_id = run_response.json()["id"]
+
+    endpoint = f"{REGISTRY_URL}/api/model_registry/v1alpha3/experiment_runs/{experiment_run_id}/artifacts"
+
+    payload = {
+        "artifactType": artifact_type,
+        "name": f"my-test-experiment-artifact-post-{secrets.randbelow(1000000)}",
+        "state": state,
+        "description": "A test experiment artifact created via direct POST test.",
+        "externalId": str(secrets.randbelow(2000000000 - 100000 + 1) + 100000)
+    }
+
+    # Add type-specific properties
+    if artifact_type == "model-artifact":
+        payload["modelFormatName"] = "tensorflow"
+        payload["modelFormatVersion"] = "1.0"
+        payload["uri"] = f"{uri_prefix}my-test-model.pkl"
+    elif artifact_type == "doc-artifact":
+        payload["uri"] = f"{uri_prefix}documentation.pdf"
+    elif artifact_type == "dataset-artifact":
+        payload["uri"] = f"{uri_prefix}dataset.parquet"
+        payload["sourceType"] = "s3"
+        payload["source"] = "s3://test-bucket/datasets/"
+    elif artifact_type == "metric":
+        payload["value"] = 0.95  # Example accuracy value
+        # Use current time in milliseconds plus random offset to ensure uniqueness
+        import time
+        current_ms = int(time.time() * 1000)
+        random_offset = secrets.randbelow(1000000)  # Random offset up to 1 million ms (1000 seconds)
+        payload["timestamp"] = str(current_ms + random_offset)
+    elif artifact_type == "parameter":
+        payload["value"] = "0.001"  # Example learning rate
+        payload["parameterType"] = "string"
+
```

Comment:
can this be moved to an utils/helper? same code is added to the previous test, lines 60-8